.slashrc

```diff
@@ -341,6 +343,42 @@ class MediaPlugin(slash.plugins.PluginInterface):
       slash.context.session.results.global_result.get_log_dir(), "results.xml")
     tree.write(filename)
 
+  def _load_blacklist(self, file_path = None):
+    if file_path is None or len(file_path) == 0 or not os.path.exists(file_path):
+      return
+
+    import json
+    for file in os.listdir(file_path):
+      if file.endswith('.json'):
+        with open(os.path.join(file_path, file), 'rb') as f:
+          black_json = json.load(f)
+          if 'blacklist' not in black_json or 'platform' not in black_json or 'driver' not in black_json:
+            continue
+          key = black_json['platform'] + ' - ' + black_json['driver']
+          if key not in self.blacklist:
+            self.blacklist[key] = {}
+          platform_driver = self.blacklist[key]
+          for suite,case in black_json['blacklist'].items():
+            if suite not in platform_driver:
+              platform_driver[suite] = {}
+            platform_driver[suite].update(**case)
```

Comment:
Instead of parsing the json after loading it from the file, just impose the expected structure in the json directly.  The key can just be a single string like "platform:driver" syntax.  Then, all we have to do is self.blacklist.update(json.load(f))